packages/perseus/src/widget-ai-utils/graded-group/graded-group.test.ts

```diff
@@ -0,0 +1,197 @@
+        // Arrange
+        const {renderer} = renderQuestion(question);
+
+        // Act
+        const json = renderer.getPromptJSON();
+
+        // Assert
+        expect(json).toEqual({
+            content:
+                "---\n\n##Check your understanding!\n\n[[☃ graded-group 1]]\n\n",
+            widgets: {
+                "graded-group 1": {
+                    type: "graded-group",
+                    title: "Metabolic strategies of bacteria",
+                    content:
+                        "1. **Which of the following statements about metabolic strategies of bacteria are true?**\n\n [[☃ categorizer 1]]",
+                    widgets: {
+                        "categorizer 1": {
+                            type: "categorizer",
+                            options: {
+                                items: [
+                                    "Some bacteria conduct photosynthesis and produce oxygen, much like plants.",
+                                    "Bacteria are always autotrophic but they may get energy from either light or chemical sources.",
+                                    "Some chemosynthetic bacteria introduce energy and fixed carbon into communities where photosynthesis is not possible (e.g., deep-sea vents).",
+                                    "Some bacteria live symbiotically inside of host organisms and provide the host with nutrients.",
+                                ],
+                                categories: ["True", "False"],
+                            },
+                            userInput: {
+                                itemToCategoryMapping: [],
+                            },
+                        },
+                    },
+                    hint: {
+                        content:
+                            "Some bacteria synthesize their own fuel molecules/fix their own carbon (autotrophic), while others take in fixed carbon from their environments (heterotrophic).\n\nSome autotrophs use light energy to synthesize their own fuel molecules, while others extract energy from chemical sources.\n\nBacteria that extract energy from chemical sources and use it to fix carbon are called chemosynthetic organisms.  These bacteria may be essential to communities where light is not available, like those around deep-sea vents. They can form the base of the food chain (act as primary producers) in these ecosystems.\n\nSome bacteria have symbiotic (mutually beneficial) relationships with other organisms, living inside these organisms and providing them with nutrients.\n\n**The following statements about the metabolic strategies of bacteria are true:**\n\n[[☃ categorizer 1]]",
+                        widgets: {},
+                    },
+                },
+            },
+        });
+    });
+
+    it("should get prompt json which matches the state of the UI when the hint is expanded", async () => {
```

Comment:
I don't see that the output varies if the hint is collapsed or expanded.  As a result, is the difference between these two tests relevant?

Reply:
So when the hint is collapsed, the widgets don't render, and [hint.widgets = {}](https://github.com/Khan/perseus/pull/1831/files#diff-d93802a06953a622491d7fb4ff6911e6e9a606700402929300694ed307f603edR128).
When expanded, [we see the rendered widget](https://github.com/Khan/perseus/pull/1831/files#diff-d93802a06953a622491d7fb4ff6911e6e9a606700402929300694ed307f603edR175-R191).